src/test/mocks.ts

```diff
@@ -96,13 +100,15 @@ export class MockWatcher implements Watcher {
 }
 
 export class MockClient extends MockWatcher implements ErdstallClient {
+	readonly tokenCache: TokenCache;
 	private readonly contract: Address;
 	private metadata: Map<string, NFTMetadata>;
 
 	constructor(contract: Address) {
 		super();
 		this.contract = contract;
 		this.metadata = new Map();
+		this.tokenCache = new TokenTypesCache();
```

Comment:
I think for the mock we actually need a `MockTokenProvider` that never queries anything and all types and holders have to be set with setters. But let's postpone the implementation of this when it's needed :)

Reply:
Agree, I did not want to create a `MockTokenProvider`, because I also deemed it not to be necessary currently. I will create an issue for this though.